go/common/l1_transaction.go

```diff
@@ -0,0 +1,167 @@
+package common
+
+import (
+	"crypto/ecdsa"
+	"fmt"
+	"math/big"
+
+	gethcommon "github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/crypto/kzg4844"
+)
+
+// L1TenTransaction is an abstraction that transforms an Ethereum transaction into a format that can be consumed more
+// easily by TEN.
+type L1TenTransaction interface{}
+
+type L1RollupTx struct {
+	Rollup EncodedRollup
+}
+
+type L1RollupHashes struct {
+	BlobHashes []gethcommon.Hash
+}
+
+type L1DepositTx struct {
+	Amount        *big.Int            // Amount to be deposited
+	To            *gethcommon.Address // Address the ERC20 Transfer was made to (always be the Management Contract Addr)
+	Sender        *gethcommon.Address // Address that issued the ERC20, the token holder or tx.origin
+	TokenContract *gethcommon.Address // Address of the ERC20 Contract address that was executed
+}
+
+type L1RespondSecretTx struct {
+	Secret      []byte
+	RequesterID gethcommon.Address
+	AttesterID  gethcommon.Address
+	AttesterSig []byte
+}
+
+type L1SetImportantContractsTx struct {
+	Key        string
+	NewAddress gethcommon.Address
+}
+
+// Sign signs the payload with a given private key
+func (l *L1RespondSecretTx) Sign(privateKey *ecdsa.PrivateKey) *L1RespondSecretTx {
```

Comment:
this is a bit confusing here. It's probably just used during testing, so it should be moved there